terraform/environments/analytical-platform-compute/iam-roles.tf

```diff
@@ -342,3 +342,44 @@ module "quicksight_vpc_connection_iam_role" {
 
   tags = local.tags
 }
+
+module "lake_formation_to_data_production_mojap_derived_tables_role" {
+  #checkov:skip=CKV_TF_1:Module registry does not support commit hashes for versions
+  #checkov:skip=CKV_TF_2:Module registry does not support tags for versions
+
+  source  = "terraform-aws-modules/iam/aws//modules/iam-assumable-role"
+  version = "5.46.0"
+
+  create_role       = true
+  role_requires_mfa = false
+
+  role_name_prefix = "lf-data-prod-mojap-derived-"
+
+  # number_of_custom_role_policy_arns = 1
+
+  custom_role_policy_arns = [
+    module.data_production_mojap_derived_bucket_lake_formation_policy.arn,
+  ]
+
+  create_custom_role_trust_policy = true
+  custom_role_trust_policy        = data.aws_iam_policy_document.custom_lake_formation_trust_policy.json
+
+  tags = local.tags
+}
+
+data "aws_iam_policy_document" "custom_lake_formation_trust_policy" {
+  statement {
+    effect = "Allow"
+    actions = [
+      "sts:AssumeRole",
+      "sts:SetContext"
+    ]
+    principals {
+      type = "Service"
+      identifiers = [
+        "glue.amazonaws.com",
+        "lakeformation.amazonaws.com"
+      ]
+    }
+  }
+}
```

Comment:
This can probably be replaced with https://github.com/terraform-aws-modules/terraform-aws-iam/blob/master/examples/iam-assumable-role/main.tf#L19-L21

Reply:
That doesn't grant the `"sts:SetContext"` action permission unfortunately.